rainier-plot/src/main/scala/com/stripe/rainier/plot/Jupyter.scala

```diff
@@ -6,125 +6,146 @@ import com.cibo.evilplot.plot._
-  }
+  def density[N](seq: Seq[N],
+                 xLabel: String = "",
+                 bounds: Option[Bounds] = None)(implicit num: Numeric[N],
+                                                theme: Theme): Plot =
+    Histogram(seq.map { n =>
+      num.toDouble(n)
+    }, binningFunction = Histogram.density, xbounds = bounds)
+      .xLabel(xLabel)
+      .yLabel("Density")
+      .xAxis()
+      .yAxis()
+      .frame()
 
-  private def scatterPlot[N](seq: Seq[(N, N)])(implicit num: Numeric[N]): Plot =
-    ScatterPlot(
-      seq.map { p =>
-        Point(num.toDouble(p._1), num.toDouble(p._2))
-      },
-      pointRenderer = Some(
-        PointRenderer.default[Point](Some(HSLA(210, 100, 56, 0.5)), Some(2))))
+  def scatter[M, N](seq: Seq[(M, N)])(implicit mNum: Numeric[M],
+                                      nNum: Numeric[N],
+                                      theme: Theme): Plot =
+    ScatterPlot(seq.map { p =>
+      Point(mNum.toDouble(p._1), nNum.toDouble(p._2))
+    })
 
-  def scatter[N](seq: Seq[(N, N)], xLabel: String = "x", yLabel: String = "y")(
-      implicit num: Numeric[N],
-      oh: OutputHandler,
-      extent: Extent): Unit =
-    show(
-      scatterPlot(seq)
-        .xAxis()
-        .yAxis()
-        .frame()
-        .xLabel(xLabel)
-        .yLabel(yLabel))
+  def contour[M, N](seq: Seq[(M, N)])(implicit mNum: Numeric[M],
+                                      nNum: Numeric[N],
+                                      theme: Theme): Plot =
+    ContourPlot(seq.map { p =>
+      Point(mNum.toDouble(p._1), nNum.toDouble(p._2))
+    }, surfaceRenderer = Some(SurfaceRenderer.contours()))
 
-  def scatter(seq: Seq[Map[String, Double]], xKey: String, yKey: String)(
-      implicit oh: OutputHandler,
-      extent: Extent): Unit =
-    scatter(seq.map { s =>
-      (s(xKey), s(yKey))
-    }, xKey, yKey)
-
-  def scatterLines(seq: Seq[Map[String, Double]], xKey: String, yKey: String)(
-      fn: Double => Seq[Double])(implicit oh: OutputHandler,
-                                 extent: Extent): Unit = {
-    val scatter = scatterPlot(seq.map { s =>
-      (s(xKey), s(yKey))
+  def line[M, N](seq: Seq[(M, N)])(implicit mNum: Numeric[M],
+                                   nNum: Numeric[N],
+                                   theme: Theme): Plot =
+    LinePlot(seq.map {
+      case (m, n) =>
+        Point(mNum.toDouble(m), nNum.toDouble(n))
     })
-    val functionPlots = fn(0.0).zipWithIndex.toList.map {
-      case (_, i) =>
-        FunctionPlot.series(x => fn(x)(i), "", HSLA(0, 100, 56, 0.5), Some(scatter.xbounds))
+
+  def line(xbounds: Bounds)(fn: Double => Double)(implicit theme: Theme): Plot =
+    lines(xbounds) { x =>
+      List(fn(x))
     }
-    show(
-      Overlay
-        .fromSeq(scatter :: functionPlots)
-        .xAxis()
-        .yAxis()
-        .frame()
-        .xLabel(xKey)
-        .yLabel(yKey))
+
+  def lines(xbounds: Bounds)(fn: Double => Seq[Double])(
```

Comment:
Not sure it's important to have a dedicated `lines` rather than generating N separate plots with `line`, but doesn't hurt so I don't have a strong opinion.

Reply:
I think it's convenient enough to be worth leaving